packages/backend/src/apps/lettersg/common/add-auth-header.ts

```diff
@@ -1,12 +1,34 @@
 import { TBeforeRequest } from '@plumber/types'
 
+import logger from '@/helpers/logger'
+
 import { getApiBaseUrl } from './api'
 
 const addAuthHeader: TBeforeRequest = async ($, requestConfig) => {
-  const apiKey = $.auth.data.apiKey as string
+  const apiKey = $.auth.data.apiKey
+
+  if (typeof apiKey !== 'string') {
+    logger.error({
+      event: 'auth-header-apikey-error',
+      error: 'LetterSG: API key must be set',
+      flowId: $.flow.id,
+    })
+    throw new Error('Missing API key')
+  }
```

Comment:
Would it be good to be more precise about the error type and logging message?
```suggestion
  if (typeof apiKey !== 'string') {
    logger.error({
      event: 'auth-header-apikey-error',
      error: `[LetterSG] Unexpected API key type: ${typeof apiKey}`,
      flowId: $.flow.id,
    })
    throw new Error(`[LetterSG] Unexpected API key type: ${typeof apiKey}`)
  }
```

Reply:
sounds good, thanks for the suggestion!